vint/linting/linter.py

```diff
@@ -112,11 +112,31 @@ def lint_file(self, path):
             decoding_error = self._create_decoding_error(path, str(exception))
             return [decoding_error]
 
+        self._traverse(root_ast, path)
+
+        return self._violations
+
+
+    def lint_text(self, text):
+        self._log_file_path_to_lint('stdin')
+
+        try:
+            root_ast = self._parser.parse(text)
+        except vimlparser.VimLParserException as exception:
+            parse_error = self._create_parse_error('stdin', str(exception))
+            return [parse_error]
+        except EncodingDetectionError as exception:
+            decoding_error = self._create_decoding_error('stdin', str(exception))
+            return [decoding_error]
```

Comment:
Would be nice to support `--stdin-filename` that could be used to configure the name (instead of `stdin`).